97.md

```diff
@@ -0,0 +1,82 @@
+
+# Recommendation Subscriptions list
+
+A special event with kind `10020` "Recommendation Subscription List" is defined as a [NIP-16 - Replaceable Events](16.md) of public and/or private `p` tags, one for each of the subscribed services the author intends to use. Each tag entry should contain the public key for the service provider and a relay URL where the provider hosts the recommendation events. i.e., `["p", <32-bytes hex key>, <main relay URL>]`.
+
+For example:
+
+```json
+{
+  "kind": 10020,
+  "tags": [
+    ["p", "91cf9..4e5ca", "wss://provider1.com/"],
+    ["p", "14aeb..8dad4", "wss://provider2.com/nostr"],
+    ["p", "612ae..e610f", "ws://provider3.com/ws"]
+  ],
+  "content": "<private list>",
+  ...other fields
+}
+```
+
+Providers can offer as many algorithms as they way as long as they offer one per public key. 
+
+# Recommendation Request
+
+Clients can request recommendations at will using a kind `20020` ephemeral event. `p` tags MUST contain the group of users to make the recommendation for as well as the pub key of the service provider. The `.content` field MUST contains an array of NOSTR `REQ` filters ([NIP-1 - Basic Protocol](1.md)) that describe the type and constraints of the requested recommendation events.
+
+```json
+{
+  "kind": 20020,
+  "tags": [
+    ["p", "<service pubkey>"],
+    ["p", "91cf9..4e5ca", "wss://relay1.com"],
+  ],
+  "content": "[<FILTER1>, <FILTER2>, ...]",
+  ...other fields
+}
+```
+
+Clients may specify several recommendation filters, i.e. `[{ "kinds": [1, 1065]}, { "#p": ["91cf9..4e5ca"], limit 200}, ...]`. Clients may include `kinds`, `tags` and other filter fields to restrict recommendation results to particular event kinds.
+
+Providers SHOULD interpret the recommendation request to the best of their ability and return events that it deems interesting to the requesting pubkey group. Recommendations can also include trending events as long as they are deemed of interest.
+
+Clients MAY send requests to any provider or to just the select few the user is interested in at the moment.
+
+# Recommendation Response
+
+Providers respond with an ephemeral event with kind `20021`. This event MUST contains the tags to the recommended event ids and the relays to find them. An optional ranking score from 0 (least interesting) to 1 (most interesting) can be added per tag. The `p` tag of the requester and the `e` tag for the requesting event SHOULD be added as well. 
+
+```json
+{
+  "kind": 20021,
+  "tags": [
+    ["p", "<Request Pub Key>"]
+	["e", "<Request Event ID>"]
+    ["e", "91cf9..4e5ca", "wss://alicerelay.com/", <score>],
+    ["e", "14aeb..8dad4", "wss://bobrelay.com/nostr", <score>],
+    ["e", "612ae..e610f", "ws://carolrelay.com/ws", <score>]
+  ],
```

Comment:
Maybe we should allow any tags, not just 'e' as a recommendation? I might wish to recommend p profiles, r urls, t hashtags etc.

Reply:
The only issue here is that there is no good way to use REQ filters to ask for only events or only URLs.

Reply:
Again, back to the question of 'type of algo' - how does a client know which filters to provide in the request? An algo might work let's say for music tracks only, how would client discover that? I'm guessing at least the appropriate kinds must be somehow discover-able by a client?

Or would it be like: this client only works with kind1, so it specifies kind1, and if results are empty - user will see empty results and will not subscribe to the list? 

Maybe there should be some other replaceable event announcing an algo? Like 'this profile is NIP-97 service and we support these kinds' type of thing, with a human-readable description? Not only could it solve the compatibility issue, but it could also allow clients to discover the algos and offer them to users for subscription?

Reply:
I was hoping to keep service discoverability events out of this PR for now, but it might be a parameter in the User Metadata for automatic services offered (obviously a larger scope than just Recommendations).

Reply:
The interface dictates the type of event. The client will just ask the subscription list for the types each part of the screen can display. If nothing comes back then it displays nothing (or a message asking the user to subscribe to a service)

Reply:
Agree on making discoverability a separate NIP, and the 'just display nothing if kind not supported' approach.